source/MQTTFileDownloader.c

```diff
@@ -335,30 +376,156 @@ static MQTTFileDownloaderStatus_t handleCborMessage( uint8_t * decodedData,
     return handleStatus;
 }
 
-static MQTTFileDownloaderStatus_t handleJsonMessage( uint8_t * decodedData,
-                                                     size_t * decodedDataLength,
-                                                     uint8_t * message,
-                                                     size_t messageLength )
+static bool getNumberFromString( const char * str,
+                                 size_t len,
+                                 int32_t * num )
```

Comment:
Feel like there should be a NULL check for this pointer?

Reply:
I see that the places this is called perform the check before calling this. Where it should be fine, just something i noticed.

Reply:
This being an internal function, the validity of parameters is already checked by the caller.